pkg/runtime/mac/mac.go

```diff
@@ -0,0 +1,159 @@
+	}
+
+	_, err = m.cliClient.Create(imageID)
+	if err != nil {
+		return api.MakeFailedUpdateStatus(unit.Name, unit.Image, "VMCreateFailed"), err
+	}
+	return api.MakeStillCreatingStatus(unit.Name, unit.Image, "VMTemplatePulled"), nil
+}
+
+func (m *MacRuntime) StartContainer(unit api.Unit, spec *api.PodSpec, podName string) (*api.UnitStatus, error) {
+	vmIDInterface, ok := m.UnitsVMIDs.Load(unit.Name)
+	if !ok {
+		return nil, fmt.Errorf("cannot find vm id for unit %s", unit.Name)
+	}
+	vmID := vmIDInterface.(string)
+	unitStatus, err := m.cliClient.Start(vmID)
+	if err != nil {
+		return nil, err
+	}
+	if unitStatus.Status != AnkaStatusOK {
+		return api.MakeFailedUpdateStatus(unit.Name, unit.Image, "VMStartFailed"), fmt.Errorf("cannot start vm: %s", unitStatus.Message)
+	}
+	started := true
+	return &api.UnitStatus{
+		Name: unit.Name,
+		State: api.UnitState{
+			Running: &api.UnitStateRunning{StartedAt: api.Now()},
+		},
+		Image:   unit.Image,
+		Ready:   true,
+		Started: &started,
+	}, nil
+}
+
+func getVMImageIDFromUnitImage(unit api.Unit) (string, error) {
+	// Let's set a convention that if container image has prefix mac-anka-img:
+	// then we'll treat suffix as anka VM Image ID.
+	// e.g. mac-anka-img:<vm-id>
+	vmImgName := strings.Split(unit.Image, ":")
+	if len(vmImgName) < 2 {
+		return "", fmt.Errorf("image has to be in format %s:vm-id", vmImagePrefix)
+	}
+	if vmImgName[0] != vmImagePrefix {
+		return "", fmt.Errorf("image has to be in format %s:vm-id", vmImagePrefix)
+	}
+	return vmImgName[1], nil
+}
+
+func (m *MacRuntime) RemoveContainer(unit *api.Unit) error {
+	// TODO
+	return nil
+}
+
+func (m *MacRuntime) ContainerStatus(unitName, unitImage string) (*api.UnitStatus, error) {
+	// TODO
+	vmIDinterface, ok := m.UnitsVMIDs.Load(unitName)
+	if !ok {
+		return nil, fmt.Errorf("cannot find vm id for unit %s", unitName)
+	}
+	vmID := vmIDinterface.(string)
+	showOutput, err := m.cliClient.Show(vmID)
+	if err != nil {
+		return nil, err
+	}
+	switch showOutput.Body.Status {
+	case "stopped":
+		// vm is stopped
+		return api.MakeFailedUpdateStatus(unitName, unitImage, "VMStopped"), nil
+	case "suspended":
+		// vm is suspended
+		return api.MakeFailedUpdateStatus(unitName, unitImage, "VMSuspended"), nil
+	case "running":
+		return &api.UnitStatus{
+			Name:  unitName,
+			State: api.UnitState{Running: &api.UnitStateRunning{}},
+			Image: unitImage,
+		}, nil
+	case "failed":
+		return api.MakeFailedUpdateStatus(unitName, unitImage, "VMFailed"), nil
+
+	}
+	return nil, nil
+}
+
+func (m *MacRuntime) GetLogBuffer(unitName string) (*logbuf.LogBuffer, error) {
+	return nil, nil
+}
+
+func (m *MacRuntime) ReadLogBuffer(unitName string, n int) ([]logbuf.LogEntry, error) {
+	return nil, nil
+}
+
+func (m *MacRuntime) UnitRunning(unitName string) bool {
+	return true
+}
+
+func (m *MacRuntime) GetPid(unitName string) (int, bool) {
+	return 0, false
+}
```

Comment:
I'm wondering if it's time to fix the leaky abstractions like GetPid(), the way log buffers are handled etc before we take the next step with podman, anka, etc. What do you think?

Reply:
Hmm, I'd go with iterative approach - and fix them once we work on podman implementation of given command, e.g. `GetPid` is used only in `exec` and `attach` handlers - so part of the work on those commands with podman should be getting rid of it. It may be tricky to remove all interface methods we don't want (GetPid, etc.) in one PR, as it will require massive changes. Let's tackle it one by one

Reply:
We can temporarily live without exec, attach and logs (though we should fix them after a first demo milestone), but network namespacing and pod IP handling is necessary right now in the update handler in `pkg/server/server.go`. Without that, dispatching a pod won't work I believe.